app/code/Magento/Theme/Controller/Result/JsFooterPlugin.php

```diff
@@ -32,34 +32,57 @@ public function __construct(ScopeConfigInterface $scopeConfig)
     }
 
     /**
-     * Put all javascript to footer before sending the response.
+     * Moves all JavaScript tags to the end of body if this feature is enabled.
      *
      * @param Http $subject
      * @return void
      */
     public function beforeSendResponse(Http $subject)
     {
-        $content = $subject->getContent();
-        $script = [];
-        if (is_string($content) && strpos($content, '</body') !== false) {
-            if ($this->scopeConfig->isSetFlag(
-                self::XML_PATH_DEV_MOVE_JS_TO_BOTTOM,
-                ScopeInterface::SCOPE_STORE
-            )
-            ) {
-                $pattern = '#<script[^>]*+(?<!text/x-magento-template.)>.*?</script>#is';
-                $content = preg_replace_callback(
-                    $pattern,
-                    function ($matchPart) use (&$script) {
-                        $script[] = $matchPart[0];
-                        return '';
-                    },
-                    $content
-                );
-                $subject->setContent(
-                    str_replace('</body', implode("\n", $script) . "\n</body", $content)
-                );
+        $content = (string)$subject->getContent();
+        $bodyEndTag = '</body';
+        $isEndBodyTagFound = strpos($content, $bodyEndTag) !== false;
+        $shouldMoveJsToBottom = $this->scopeConfig->isSetFlag(
+            self::XML_PATH_DEV_MOVE_JS_TO_BOTTOM,
+            ScopeInterface::SCOPE_STORE
+        );
+
+        if ($isEndBodyTagFound && $shouldMoveJsToBottom) {
+            if ($scripts = $this->extractScriptTags($content)) {
```

Comment:
Creating new variable inside for statement isn’t good thing as it quite often causes not clear understanding. Could you extract it to separate line, like this?
```suggestion
            $scripts = $this->extractScriptTags($content)
            if ($scripts) {
```

Reply:
Good catch in both cases, I'll adjust the code.